### DISCUSSION TITLE
Can't unobserve

### DISCUSSION THREAD

**Discussion by maxpowel** [JuulLabs/kable#642] [Q&A] [answered]
Hello, my doubt is similar to this https://github.com/JuulLabs/kable/discussions/592, but in my case the observe is not canceled even when I cancel the flow.

This is my code:

```
viewModelScope.launch {
            withContext(Dispatchers.IO) {

            val t = p.observe(characteristicOf(
                            service="0000ff11-cc7b-662b-984b-7c34a9bc82d0",
                            characteristic = "0000ff11-cc7b-662b-984b-7c34a9bc82d5"
                        )).cancellable()
            
            val tjob = this.launch {

                 t.onCompletion {
                    Log.d("CICI", "COMPLETO")
                } 
                t.onEach { value ->
                    Log.d("LOG", "reading value of size ${value.size}")
                    
                    if (elems.size == 10) {
                        // All elements read
                        cancel()
                    }
                }.collect()
            }
            tjob.join()
            Log.d("LOG", "Here is expect set notification to false")
   }
}
```

I receive all my data but the "unobserve" never happens. `onCompletion` is never called either.
I tried many ways to consume the flow and different types of coroutinescope without success.

Can you help me to find what I am doing wrong?

Thank you

**Comment by twyatt** [comment]
Are you seeing your "Here is expect set notification to false" log?

**Comment by maxpowel** [comment, marked as answer]
Thank you for your interest. Looks that the nrf-connect application were affecting. It is a tool to debug bluetooth connections and it was affecting to any bluetooth conection just for being opened.

For anyone else having troubles check other applications because android is very picky with applications using bluetooth.

Thanks

**Reply by twyatt** [reply]
Thanks for following up and sharing your findings. 💯
